pkg/build/linter.go

```diff
@@ -141,12 +167,52 @@ func worldWriteableLinter(_ LinterContext, path string, d fs.DirEntry) error {
 	return nil
 }
 
+func emptyPostLinter(lctx LinterContext, fsys fs.FS) error {
+	foundfile := false
+	walkCb := func(path string, _ fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if path == "." {
+			// Skip root
+			return nil
+		}
+
+		foundfile = true
+		return fs.SkipAll
+	}
+
+	err := fs.WalkDir(fsys, ".", walkCb)
+	if err != nil {
+		return err
+	}
+
+	// Nothing to do
+	if foundfile {
+		return nil
+	}
+
+	options, err := lctx.getPackageOptions()
+	if err != nil {
+		return err
+	}
+
+	if options.NoProvides {
```

Comment:
I would prefer a direct signal (such as disabling the linter) rather than magical thinking here.

Reply:
Done.